Resources/public/js/views/plugins/cof-createcontent-universaldiscoveryserviceplugin.js

```diff
@@ -122,6 +125,116 @@ YUI.add('cof-createcontent-universaldiscoveryserviceplugin', function (Y) {
              */
             app.fire('contentDiscover');
         },
+
+        /**
+         * Loads the content type data.
+         *
+         * @protected
+         * @method _loadContentTypeData
+         * @param event {Object} event facade
+         */
+        _loadContentTypeData: function (event) {
+            var type = event.contentType,
+                host = this.get('host');
+
+            type.load({api: host.get('capi')}, Y.bind(function (error) {
+                if (error) {
+                    /**
+                     * Displays a notification bar with error message.
+                     * Listened by eZ.PlatformUIApp
+                     *
+                     * @event notify
+                     * @param notification {Object} notification data
+                     */
+                    host.fire('notify', {
+                        notification: {
+                            text: "Could not load the content type with id '" + type.get('id') + "'",
+                            identifier: 'loading-content-type-error',
+                            state: 'error',
+                            timeout: 0
+                        }
+                    });
+
+                    return;
+                }
+
+                this._setContentTypeData(event);
+            }, this));
+        },
+
+        /**
+         * Sets the data required to create new content
+         *
+         * @method _setContentTypeData
+         * @protected
+         * @param event {Object} event facade
+         */
+        _setContentTypeData: function (event) {
+            var content = new Y.eZ.Content(),
+                version = new Y.eZ.Version(),
+                type = event.contentType,
+                mainLanguageCode = type.get('mainLanguageCode'),
+                host = this.get('host'),
+                user = host.get('app').get('user'),
+                target = event.target,
+                defaultFields = {};
+
+            content.set('name', 'New "' + type.get('names')[mainLanguageCode] + '"');
+
+            Y.Object.each(type.get('fieldDefinitions'), function (fieldDef, identifier) {
+                defaultFields[identifier] = {
+                    fieldDefinitionIdentifier: identifier,
+                    fieldValue: fieldDef.defaultValue,
+                };
+            });
+
+            host.set('content', content);
```

Comment:
niptick: you can use host.setAttrs({content: content, languageCode: mainLanguageCode, contenTy.....}